src/action/PythonAction.cpp

```diff
@@ -97,31 +97,43 @@ void PythonAction::performAction(double time)
   PyObject *dataArgs   = PyTuple_New(_numberArguments);
   PyObject *pythonTime = PyFloat_FromDouble(time);
   PyTuple_SetItem(dataArgs, 0, pythonTime);
-  if (_sourceData) {
-    npy_intp sourceDim[]  = {_sourceData->values().size()};
-    double * sourceValues = _sourceData->values().data();
-    // PRECICE_ASSERT(_sourceValues == NULL);
-    _sourceValues = PyArray_SimpleNewFromData(1, sourceDim, NPY_DOUBLE, sourceValues);
-    PRECICE_CHECK(_sourceValues != nullptr, "Creating python source values failed. Please check that the source data name is used by the mesh in action:python.");
-    PyTuple_SetItem(dataArgs, 1, _sourceValues);
-  }
-  if (_targetData) {
+
+  int i = 0;
+  for (auto &targetStample : _targetData->stamples()) { // iterate over _targetData, because it must always exist
+    PRECICE_ASSERT(_targetData);                        // _targetData is mandatory, cannot call setSampleAtTime, if target data is not provided!
+
+    if (_sourceData) {                                     // _sourceData is optional
+      auto &sourceStample   = _sourceData->stamples()[i];  // simultaneously iterate over _targetData->stamples()
+      _sourceData->values() = sourceStample.sample.values; // put data into temporary buffer
+      PRECICE_ASSERT(sourceStample.timestamp == targetStample.timestamp, "Time mesh of source and target stample must agree!");
```

Comment:
Let's better make this one a "check".
And don't we need to floating point comparison here?

Reply:
Done. 9aa3908000c4b714cea9b920d6c7bfe7a59f92f9